cms/djangoapps/contentstore/asset_storage_handlers.py

```diff
@@ -88,6 +88,52 @@ def handle_assets(request, course_key_string=None, asset_key_string=None):
     return HttpResponseNotFound()
 
 
+def check_asset_pre_upload(request, course_key_string):
+    """
+    Check if the files have any filename conflicts with existing assets.
+
+    Returns a JSON response with the following fields:
+    - allowed: boolean, whether the user is allowed to upload an asset
+    - reason: string, the reason why the user is not allowed to upload an asset
+    - files: list of strings, the filenames that already exist
+    """
+    course_key = CourseKey.from_string(course_key_string)
+    if not has_course_author_access(request.user, course_key):
+        raise PermissionDenied()
+
+    result = _get_pre_existing_filenames(request, course_key)
```

Comment:
Please set some limit on the number of files that can be queried against at one time by erroring out immediately if the number of filesnames to check exceeds that limit. 100 maybe? It should be high enough that there's no reasonable use case where it would be hit by a human user. But low enough where it won't bring down the entire process with a timeout trying to pull that much data from MongoDB.

Reply:
Have limited the max upload size to 100 on https://github.com/openedx/edx-platform/pull/33493 , which is controlled by the pageSize in the updated PR